Changes to allow for all scans to complete

kdelee opened the pull request:
Several changes to allow all the scans that happen at the beginning of the test
session to complete before the test suite proceeds. This includes allowing
longer boot times (but we proceed if the machine is responsive), and longer
wait times for a scan to complete since it may be queued behind several other
scans.

Additionally, we provide a fresh client to each test so that it has a current
authentication token.

Closes #223

kdelee committed:
Changes to allow for all scans to complete

Several changes to allow all the scans that happen at the beginning of the test
session to complete before the test suite proceeds. This includes allowing
longer boot times (but we proceed if the machine is responsive), and longer
wait times for a scan to complete since it may be queued behind several other
scans.

Additionally, we provide a fresh client to each test so that it has a current
authentication token and make sure the clients of objects about to be deleted
have a good token.

Closes #223

kdelee commented:
I'm hoping that this change will also alleviate some other problems we are having on Jenkins that I think are coming up because the server is busy running scans and writing to the DB.

elyezer commented on camayoc/tests/qpc/api/v1/utils.py:
```diff
@@ -157,11 +161,13 @@ def wait_until_state(scanjob, timeout=120, state='completed'):
     while (not scanjob.status() or not scanjob.status()
             == state) and timeout > 0:
-        if state == 'stopped' and scanjob.status() in QPC_SCAN_TERMINAL_STATES:
+        if state in stopped_states and scanjob.status() in stopped_states:
```
We are calling `scanjob.status()` many times here which is going a read request. I think we can do better and check the status once at the beginning of the while loop.

kdelee replied:
I think I'd like to check once at top and once after the call to time.sleep

elyezer commented on camayoc/tests/qpc/api/v1/utils.py:
```diff
@@ -157,11 +161,13 @@ def wait_until_state(scanjob, timeout=120, state='completed'):
             return
         time.sleep(5)
         timeout -= 5
+        if scanjob.status() == state:
```
This condition is already on the while loop condition, I think we should either check inside the loop of on the loop condition.

elyezer reviewed (changes requested):
LGTM, left a comment about the logic on the wait_for_state function, can you take a look on it?

kdelee committed:
Clean up wait_until_state to minimize http calls

Makes the actual pause in wait_until_state happen at the end of the loop so we
can minimize how many times we are checking the status of the job.

kdelee commented:
@elyezer I cleaned up the logic so we don't have to check multiple times in the loop by moving the call to time.sleep to the end of the loop

elyezer reviewed (approved):
LGTM
